pkg/portableresources/renderers/dapr/generic.go

```diff
@@ -56,9 +57,24 @@ func ConstructDaprGeneric(daprGeneric DaprGeneric, namespace string, componentNa
 	// Dapr specs: https://docs.dapr.io/reference/components-reference/
 	yamlListItems := []any{} // K8s fake client requires this ..... :(
 	for k, v := range daprGeneric.Metadata {
+
 		yamlItem := map[string]any{
-			"name":  k,
-			"value": v,
+			"name": k,
+		}
+
+		switch value := v.(type) {
+
+		// Nested object, has to be a secret reference
+		case map[string]any:
+			if secretKeyRef, ok := value["secretKeyRef"]; ok {
+				yamlItem["secretKeyRef"] = secretKeyRef
+			} else {
+				return unstructured.Unstructured{}, v1.NewClientErrInvalidRequest(fmt.Sprintf("Invalid metadata value for key %s in Dapr component %s", k, componentName))
+			}
```

Comment:
Could there be a case where it is not a secret reference even if it is a nested object? Do we need to do an additional check for this case to make sure that it is a secret reference?

Reply:
You can see the schema of `Component` here: https://github.com/dapr/dapr/blob/master/charts/dapr/crds/components.yaml#L48

Note that this is the **output** we're generating. 

The reason why I'm mention it is that it shows what the schema is in Dapr that we're trying roughly match. The allowed keys are:

- `value`
- `secretKeyRef`
- `envRef`

`envRef` isn't relevant for us - it's a dev-time feature.